src/internal/operators/distinct.ts

```diff
@@ -74,6 +75,11 @@ export function distinct<T, K>(keySelector?: (value: T) => K, flushes?: Observab
       })
     );
 
-    flushes?.subscribe(createOperatorSubscriber(subscriber, () => distinctKeys.clear(), noop));
+    let flush$ = flushes as InteropObservable<any> | ObservableInput<any> | undefined | any;
+    if (flushes && isInteropObservable(flushes)) {
+      flush$ = fromInteropObservable(flushes);
+    }
+
+    flush$?.subscribe(createOperatorSubscriber(subscriber, () => distinctKeys.clear(), noop));
```

Comment:
Hey @jeremymwells, thanks for the PR. Instead of using `fromInteropObservable`, you should be using `innerFrom`, like this:

```suggestion
    flushes && innerFrom(flushes).subscribe(createOperatorSubscriber(subscriber, () => distinctKeys.clear(), noop));
```